butterknife-annotations/src/main/java/butterknife/BindView.java

```diff
@@ -13,9 +13,28 @@
  * <pre><code>
  * {@literal @}BindView(R.id.title) TextView title;
  * </code></pre>
+ *
+ * If used in a library project where the fields are not final, use the other attributes:
+ * <pre><code>
+ * {@literal @}BindView(idName = "title") TextView title;
```

Comment:
That can cause an issue, but I think this is the closest you can get to being safe without being able to reference the ID directly. I would imagine swapping the IDs of two views would be rare enough that it won't cause many issues.